### ISSUE TITLE
What are bagging temperature and random strength?

### ISSUE THREAD

**Issue by fantauzzi** [catboost/catboost#373]
Problem: Documentation doesn't explain what parameters bagging_temperature and random_strength do.
catboost version: 0.8.1
Operating System: N/A
CPU: N/A
GPU: N/A

I have also studied the paper [CatBoost: unbiased boosting with categorical features](https://arxiv.org/abs/1706.09516) but couldn't recognize those parameters there. Perhaps they are called in a different way in the paper?

**Comment by echo66** [comment]
I also have the same question.

**Comment by Oktai15** [comment]
@echo66 this issue doesn't cover explanation of bagging_temperature

**Comment by annaveronika** [comment]
We'll add better documentation soon.

**Comment by annaveronika** [comment]
I'll post the answer here, it will go to the docs soon.

random_strength
This parameter helps to overcome overfitting of the model.

When selecting a new split, each possible split gets a score (for example, by how much does adding this split improve the loss function on train). After that all scores are sorted and a split with the highest score is selected.
The scores are not random. This parameter adds a normally distributed random variable to the score of the feature. It has zero mean and variance that is larger in the start of the training and decreases during the training. random_strength is the multiplier of the variance.

Using randomness during split selection procedure helps to overcome overfitting and increase the resulting quality of the model.

bagging_temperature
This parameter is responsible for Bayesian bootstrap.
Bayesian bootstrap is used by default in classification and regression modes. In ranking we use Bernoulli bootstrap by default.

In bayesian bootstrap each object is assigned random weight. If bagging temperature is equal to 1 then weights are sampled from exponential distribution. If bagging temperature is equal to 0 then all weights are equal to 1. By changing this parameter from 0 to +infty you can controll intensity of the bootstrap.

**Comment by annaveronika** [comment]
Closing the issue, feel free to reopen if you have further questions.

**Comment by m-dz** [comment]
Hi @annaveronika , I have a quick question related to the documentation and these two parameters, what are the default values for them? Looking at [docs](https://tech.yandex.com/catboost/doc/dg/concepts/python-reference_catboostregressor-docpage/) it seems to be `None`, but what that exactly means?

**Comment by annaveronika** [comment]
There is a section where each parameter is described. It contains all default values
https://tech.yandex.com/catboost/doc/dg/concepts/python-reference_parameters-list-docpage/

**Comment by m-dz** [comment]
Thank you, coundn't find this particular part of the docs!

**Comment by beltonn** [comment]
Hi @annaveronika, I also have a quick question about how feature interaction strength is calculated. From looking at the formula, I assume that it is comparing the values of a leaf that had a previous branch with the interaction and leafs who did not. I don't understand how this works when there is oblivious trees because I thought that every split is the same so all leafs would contain the interaction. 

Also, do the feature combinations have a part in the feature interaction strength calculation? Thank you!

**Comment by MariaLavrovskaya** [comment]
Hi @annaveronika, I am slightly confused by the Bayesian bootstrap procedure. In the Bayesian Bootstrap procedure that is outlined in D. Rubin “The Bayesian Bootstrap”, 1981, Section 2 the values are sampled from the uniform distribution, which is same thing as stated on the docs page https://catboost.ai/docs/concepts/algorithm-main-stages_bootstrap-options.html . However, by controlling the bagging temperature in cases where the value is not 0 or 1, from which distribution would be the weights drawn? In other words, what do you mean by bootstrap intensity?
